browser/ui/views/infobars/brave_confirm_infobar.cc

```diff
@@ -187,6 +187,22 @@ void BraveConfirmInfoBar::Layout(PassKey) {
   link_->SetPosition(gfx::Point(GetEndX() - link_->width(), OffsetY(link_)));
 }
 
+void BraveConfirmInfoBar::ViewHierarchyChanged(
+    const views::ViewHierarchyChangedDetails& details) {
+  InfoBarView::ViewHierarchyChanged(details);
+
+  // Upstream doesnt need this reorder because all other children
+  // except icon and close button are in |content_container_| that lives
+  // between both. It guarantees that close button is always the last
+  // children. However, we just add children directly instead of
+  // that container.
+  // TODO(https://github.com/brave/brave-browser/issues/48822): Add our children
+  // into content container.
+  if (details.is_add && (details.child == this) && close_button_) {
+    ReorderChildView(close_button_, children().size());
```

Comment:
How about reordering close button at the end of constructor, where all child views are attached?

Reply:
Can't do it because we have another subclass of `BraveConfirmInforBar`.

Reply:
Oops, sorry. I was confused. we don't have its subclass :) 
We could do it that way.

Reply:
Fixed.

Reply:
Thanks :)